From 6ca894351fd1ebce37414d01ade033269609bbaa Mon Sep 17 00:00:00 2001
From: denglei <denglei@eswincomputing.com>
Date: Wed, 11 Dec 2024 14:32:12 +0800
Subject: [PATCH 334/416] WIN2030-16445:fix:fix the issue of audio recording.

Changelogs:
	1.fix the issue of audio recording on Z530 board.
	2.open the es5340 npu regulator.

Change-Id: Ia788618f8d6c937e9526acd00105cc1c4a8a63e4
Signed-off-by: denglei <denglei@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7700-z530.dts |  1 +
 arch/riscv/configs/eic7700_defconfig       |  1 +
 drivers/regulator/es5340.c                 |  2 --
 sound/soc/codecs/eswin/es8328.c            | 10 +++++++---
 4 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7700-z530.dts b/arch/riscv/boot/dts/eswin/eic7700-z530.dts
index 4059881343d1..2926d780b79c 100644
--- a/arch/riscv/boot/dts/eswin/eic7700-z530.dts
+++ b/arch/riscv/boot/dts/eswin/eic7700-z530.dts
@@ -137,6 +137,7 @@ d0_es8388_0: es8388-0@10 {
 		compatible = "eswin,es8388";
 		reg = <0x10>;
 		#sound-dai-cells = <0>;
+		eswin-plat = <3>;
 		port {
 			d0_codec0_endpoint: endpoint {
 				system-clock-frequency = <12288000>;
diff --git a/arch/riscv/configs/eic7700_defconfig b/arch/riscv/configs/eic7700_defconfig
index e3a7b779ee59..353c29d85325 100644
--- a/arch/riscv/configs/eic7700_defconfig
+++ b/arch/riscv/configs/eic7700_defconfig
@@ -479,6 +479,7 @@ CONFIG_WATCHDOG=y
 CONFIG_DW_WATCHDOG=y
 CONFIG_REGULATOR=y
 CONFIG_REGULATOR_ES5340=y
+CONFIG_REGULATOR_MPQ8785=y
 CONFIG_REGULATOR_PCA9450=y
 CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_SUPPORT_FILTER=y
diff --git a/drivers/regulator/es5340.c b/drivers/regulator/es5340.c
index 359f9a6ead15..758e3b2fac54 100644
--- a/drivers/regulator/es5340.c
+++ b/drivers/regulator/es5340.c
@@ -711,9 +711,7 @@ static struct regulator_desc es5340_regulator_desc = {
 static s32 es5340_init_data(struct ES5340_DRIVER_DATA *data,
 							const struct regulation_constraints *constraints, u32 default_voltage)
 {
-	u8 value = 0;
 	s32 ret = 0;
-	u16 new_value = 0;
 	struct device *dev = &data->client->dev;
 
 	dev_info(dev,
diff --git a/sound/soc/codecs/eswin/es8328.c b/sound/soc/codecs/eswin/es8328.c
index 1ceb706a9f9a..bc05b82f5c79 100644
--- a/sound/soc/codecs/eswin/es8328.c
+++ b/sound/soc/codecs/eswin/es8328.c
@@ -101,8 +101,9 @@ static const char * const supply_names[ES8328_SUPPLY_NUM] = {
 		SNDRV_PCM_FMTBIT_S24_3LE | \
 		SNDRV_PCM_FMTBIT_S32_LE)
 
-#define EVB_BOARD  1
-#define DVB_BOARD  2
+#define EVB_BOARD   1
+#define DVB_BOARD   2
+#define Z530_BOARD  3
 
 struct es8328_priv {
 	struct regmap *regmap;
@@ -902,7 +903,7 @@ static int es8328_set_dai_fmt(struct snd_soc_dai *codec_dai,
 	/* Set MIC PGA Volume */
 	snd_soc_component_write(component, ES8328_ADCCONTROL1, 0x88);
 
-	if (es8328->eswin_plat == 2) {
+	if (es8328->eswin_plat == DVB_BOARD) {
 		if (gpiod_get_value(es8328->front_jack_gpio) == 1 && gpiod_get_value(es8328->back_jack_gpio) == 0) {
 			/* Select default capture path ---> LIN1 */
 			snd_soc_component_write(component, ES8328_ADCCONTROL2, 0);
@@ -910,6 +911,9 @@ static int es8328_set_dai_fmt(struct snd_soc_dai *codec_dai,
 			/* Select default capture path ---> LIN2 */
 			snd_soc_component_write(component, ES8328_ADCCONTROL2, 0x50);
 		}
+	} else if (es8328->eswin_plat == Z530_BOARD) {
+		/* Select default capture path ---> LIN1 */
+		snd_soc_component_write(component, ES8328_ADCCONTROL2, 0);
 	} else {
 		/* Select default capture path ---> phone mic */
 		snd_soc_component_write(component, ES8328_ADCCONTROL2, 0xf0);
-- 
2.48.1

